tidy3d/components/spice/analysis/dc.py

```diff
@@ -0,0 +1,61 @@
+"""
+This class defines standard SPICE electrical_analysis types (electrical simulations configurations).
+"""
+
+from typing import Optional
+
+import pydantic.v1 as pd
+
+from tidy3d.components.base import Tidy3dBaseModel
+from tidy3d.components.spice.sources.dc import MultiStaticTransferSourceDC
+
+
+class OperatingPointDC(Tidy3dBaseModel):
+    """
+    Equivalent to Section 11.1.2 in the ngspice manual.
+    """
+
+
+class TransferFunctionDC(Tidy3dBaseModel):
+    """This class sets parameters used in DC simulations.
+
+    Ultimately, equivalent to Section 11.3.2 in the ngspice manual.
+
+    Example
+    -------
+    TODOUPDATE Example.
+    >>> import tidy3d as td
+    >>> dc_spec = td.TransferFunctionDC(dv=0.1)
+
+    This class sets some Charge tolerance parameters.
+
+    Example
+    -------
+    >>> import tidy3d as td
+    >>> charge_settings = td.ChargeToleranceSpec(abs_tol=1e8, rel_tol=1e-10, max_iters=30)
+    """
+
+    sources: Optional[MultiStaticTransferSourceDC] = []  # todo accept a single source
```

Comment:
I'm unsure about having sources here. Every parameter except this one is related to convergence. Maybe we could have a convergence parameters class and so this transfer function only accepts sources?

Reply:
So I'm imagining this as a standard electrical DC simulation (and trying to match it as much as possible to the SPICE standard defintions because that will streamline integration).

In my mind, this class is not a simulation setting: it is a simulation analysis setting. This means that this could be treated as the `independent` variable part of the transfer function, and as such, there is a strong relationship to the DC sweep iteration parameters related to convergence.

I'm following a standard definition for SPICE DC transfer functions:

> The following options control properties pertaining to DC and OP (operating point) analyses
> and algorithms. Since transient analysis (11.1.4) is based on OP, many of the options affect
> transient simulation as well. AC analysis (11.1.3) can be performed only when a stable operat-
> ing point has been found.
> ABSTOL=x resets the absolute current error tolerance of the program. The default value is 1
> pA.
> GMIN=x resets the value of GMIN, the minimum conductance allowed by the program. The
> default value is 1.0e-12.
> GMINSTEPS=x [*] sets the number of Gmin steps to be attempted. If the value is set to zero,
> the standard gmin stepping algorithm is skipped. The standard behavior is that gmin
> stepping is tried before going to the source stepping algorithm.
> 
> 1.3.2 .DC: DC Transfer Function
> General form:
> .dc srcnam vstart vstop vincr [src2 start2 stop2 incr2]
> Examples:
> .dc VIN 0.25 5.0 0.25
> .dc VDS 0 10 .5 VGS 0 5 1
> .dc VCE 0 10 .25 IB 0 10u 1u
> .dc RLoad 1k 2k 100
> .dc TEMP -15 75 5
> The .dc line defines the dc transfer curve source and sweep limits (with capacitors open and
> inductors shorted). srcnam is the name of an independent voltage or current source, a resistor,
> or the circuit temperature. vstart, vstop, and vincr are the starting, final, and incrementing
> values, respectively. The first example causes the value of the voltage source VIN to be swept
> from 0.25 Volts to 5.0 Volts with steps of 0.25 Volt. A second source (src2) may optionally
> be specified with its own associated sweep parameters. In such a case the first source is swept
> over its own range for each value of the second source. This option is useful for obtaining
> semiconductor device output characteristics. See the example on transistor characterization

Reply:
Another question could be what do you envision having here. If we don't know maybe we could remove this field? 

As mentioned in another comment  `MultiStaticTransferSourceDC`  is another convergence parameter so, for the time being, I'd remove this field, add `step` here, and I'd add a `convergence` field that accepts a convergence class object, i.e.,
```
class TransferFunctionDC(Tidy3dBaseModel):
    convergence_settings: SteadyConvergenceSettings = pd.Field(
        SteadyConvergenceSettings(), title="...", descriptio="..."
    )
```

Reply:
Like this is how DC simualtions are normally run anyway. This is a good example that shows how it scales https://www.youtube.com/watch?v=6QzTZDm0guM

Reply:
So basically as long as we can extend the data to be compatible with the spice simulation format in https://ngspice.sourceforge.io/docs/ngspice-43-manual.pdf then it'd be fine imo.